File: typescript/src/state/concrete-state-a.ts
import { ConcreteStateB } from "./concrete-state-b.ts";
import type { Context } from "./context.ts";
import type { State } from "./state.ts";

export class ConcreteStateA implements State {
  handle(context: Context): void {
    context.setState(new ConcreteStateB());
  }

  name(): string {
    return "ConcreteStateA";
  }
}

File: python/src/patterns/state/concrete_state_a.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .context import Context


class ConcreteStateA:
    def handle(self, context: Context) -> None:
        # Local import: the two concrete states transition to each other.
        from .concrete_state_b import ConcreteStateB

        context.set_state(ConcreteStateB())

    def name(self) -> str:
        return "ConcreteStateA"
